src/main/java/com/basho/riak/client/core/RiakNode.java

```diff
@@ -877,6 +901,21 @@ public void onException(Channel channel, final Throwable t)
         }
     }
 
+    private void checkNetworkAddressCacheSettings()
+    {
+        final String property = Security.getProperty("networkaddress.cache.ttl");
+
+        int cacheTTL = property == null ? -1 : Integer.parseInt(property);
+
+        if (cacheTTL == -1)
+        {
+            logger.warn(
+                    "networkaddress.cache.ttl is currently set to cache DNS lookups forever. If you use domain names " +
+                    "for the Riak server connection, and an IP address changes (riak host or a load balancer), " +
+                    "this will block the client from creating connections with the new IP addresses.");
```

Comment:
How about:

"The java.security.networkaddress.cache.ttl property may be set to cache DNS lookups forever. Using domain names for Riak nodes or an intermediate load balancer could result in stale IP addresses being used for new connections, causing connection errors. If your JVM is using a security manager, please set the property to a value greater than zero".

According to the AWS docs, this setting only applies when a security manager is in use. So, if the property is `null`, could that mean no security manager is in use? Or should this method be used?

https://docs.oracle.com/javase/tutorial/essential/environment/security.html

We could even include a link to a wiki page in riak-java-client. What do you think?